dotnet/samples/KernelSyntaxExamples/Example76_StronglyTypedKernelResult.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Threading.Tasks;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel;
+using System.Text.Json;
+using System.Diagnostics;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Abstractions;
+using Examples;
+
+// The following example shows how to receive the results from the kernel in a strongly typed object
+// wich stores the usage in tokens and converts the JSON result to a strongly typed object, where a validation can also
+// be performed
+public class Example76_StronglyTypedKernelResult : BaseTest
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        this.WriteLine("======== LLMPrompts ========");
+
+        string openAIModelId = TestConfiguration.OpenAI.ChatModelId;
+        string openAIApiKey = TestConfiguration.OpenAI.ApiKey;
+
+        if (openAIApiKey == null)
+        {
+            this.WriteLine("OpenAI credentials not found. Skipping example.");
+            return;
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddOpenAIChatCompletion(
+                modelId: openAIModelId,
+                apiKey: openAIApiKey)
+            .Build();
+
+        var promptTestDataGeneration = "Return a JSON with an array of 3 JSON objects with the following fields: First, an id field with a random GUID, next a name field with a random company name and last a description field with a random short company description. Ensure the JSON is valid and it contains a JSON array named testcompanies with the three fields.";
+
+        // Time it
+        var sw = new Stopwatch();
+        sw.Start();
+
+        FunctionResult functionResult =
+            await kernel.InvokePromptAsync(promptTestDataGeneration);
+
+        // Stop the timer
+        sw.Stop();
+
+        var functionResultTestDataGen = new FunctionResultTestDataGen(
+            functionResult!,
+            sw.ElapsedMilliseconds);
+
+        this.WriteLine($"Test data: {functionResultTestDataGen.Result} \n");
+        this.WriteLine($"Milliseconds: {functionResultTestDataGen.ExecutionTimeInMilliseconds} \n");
+        this.WriteLine($"Total Tokens: {functionResultTestDataGen.TokenCounts!.TotalTokens} \n");
+    }
+
+    public Example76_StronglyTypedKernelResult(ITestOutputHelper output) : base(output)
+    {
+    }
+}
+
+// The structure to put the JSON result in a strongly typed object
+public class RootObject
```

Comment:
Since all classes belong to this specific example, let's make them `private sealed`.

Reply:
agree, except FunctionResultExtended ;)